File: C++/1.4.2.cpp
#include <iostream>
#include <vector>
#include <iomanip>

using namespace std;

int main() {
    int n, countXi = 0;

    // Вводим количество элементов массивов
    cout << "Enter the number of elements: ";
    cin >> n;

    // Проверка на корректность количества элементов
    if (n <= 0) {
        cout << "The number of elements must be greater than 0" << endl;
        return 1;
    }

    // Инициализируем массивы
    vector<double> X(n), Y(n), T(n);

    // Вводим данные для массивов X и Y
    for (int i = 0; i < n; i++) {
        cout << "Enter element X" << i + 1 << ": ";
        cin >> X[i];
        cout << "Enter element Y" << i + 1 << ": ";
        cin >> Y[i];
    }

    // Цикл для вычисления массива T и подсчета элементов, которые получили значения из массива X
    for (int i = 0; i < n; i++) {
        if (X[i] > Y[i]) {
            T[i] = X[i];
            countXi++;
        } else {
            T[i] = Y[i];
        }
    }

    // Выводим массив T
    cout << "Array T:" << endl;
    for (int i = 0; i < n; i++) {
        cout << "T" << i + 1 << ": " << fixed << setprecision(4) << T[i] << endl;
    }

    // Выводим количество элементов, которые получили значения из массива X
    cout << "Amount of elements, which received values from array X: " << countXi << endl;

    return 0;
}

File: pascal/1.4.2.pas
uses
  Math;

var
  n, i, countXi: Integer;
  X, Y, T: array of Real;

begin
  // Вводим количество элементов массивов
  Write('Enter amount of elements in array: ');
  ReadLn(n);

  // Инициализируем массивы
  SetLength(X, n);
  SetLength(Y, n);
  SetLength(T, n);

  // Вводим данные для массивов X и Y
  for i := 0 to n - 1 do
  begin
    Write('Enter element X', i + 1, ': ');
    ReadLn(X[i]);
    Write('Enter element Y', i + 1, ': ');
    ReadLn(Y[i]);
  end;

  countXi := 0;

  // Цикл для вычисления массива T и подсчета элементов, которые получили значения из массива X
  for i := 0 to n - 1 do
  begin
    if X[i] > Y[i] then
    begin
      T[i] := X[i];
      Inc(countXi);
    end
    else
      T[i] := Y[i];
  end;

  // Выводим массив T
  WriteLn('Array T:');
  for i := 0 to n - 1 do
    WriteLn('T', i + 1, ': ', T[i]:0:4);

  // Выводим количество элементов, которые получили значения из массива X
  WriteLn('Amount of elements, which received values from array X: ', countXi);
end.